content-services/content-sources-backend
# Fixes 1471: Integrate snapshot task into application

## Summary

This adds snapshotting to repositories, integrated with the api and tasking system.  When creating a repository, if you pass snapshot=true, a task is spawned to snapshot the repository.  This snapshot task should save information within the payload such as sync task href, publication task href, etc..., such that if the worker dies, the task should be able to be resumed.

## Testing steps

Run a create or bulk_create request as normal, but add the parameter 'snapshot=true'.

Try killing the worker (with ctrl-c or kill -9) and restarting it.  In all cases you should be able to see the task resumed and a snapshot created.  For now you'll have to check the db to see if snapshot got created:

```select distribution_path from snapshots;```

you can then curl the pulp distribution path:

```curl  localhost:8080/pulp/content/73e2d9fa-387d-4df1-ac58-fe898aebb30f/f4e2f625-4cba-4d5c-9ea5-3de353bb72f1/```

For QE:

We'll need to update automation to support the snapshot parameter, but we can't really test snapshots yet, as there is no way to monitor the snapshot or confirm its created.
